Empire/Empire.cpp

```diff
@@ -3372,9 +3372,11 @@ void Empire::ResetMeters() {
 }
 
 void Empire::UpdateOwnedObjectCounters() {
-    // ships of each species and design
+    // ships of each species and design, parts by type and class
     m_species_ships_owned.clear();
     m_ship_designs_owned.clear();
+    m_ship_parts_owned.clear();
```

Comment:
Add a comment explaining the different containers.... one indexed by part name, one by part class, summing either could be used for total ship parts count.